### ISSUE TITLE
`sort-keys` comparing `default` with parent property

### ISSUE THREAD

**Issue by QingqiShi** [facebook/stylex#391]
### Describe the issue

The new `sort-keys` eslint rule seems to give some unexpected errors, suggesting it's comparing the inner keys with parent keys.

![image](https://github.com/facebook/stylex/assets/6232457/71bc6a85-4c99-4dca-9bd2-feef7ac49871)

### Expected behavior

It's expected that the test case below is valid, however it errors with the message

> StyleX property key "default" should be above "maxWidth".

### Steps to reproduce

`@stylexjs/eslint-plugin@0.5.1`

- Add new test case in `packages/eslint-plugin/__tests__/stylex-sort-keys-test.js` with the below test case as a valid case
- Run `npm run test -w @stylexjs/eslint-plugin` and observe test failure with message `StyleX property key "default" should be above "maxWidth"`

### Test case

```js
      import stylex from 'stylex';
      const styles = stylex.create({
        nav: {
          maxWidth: {
            default: "1080px",
            ["@media (min-width: 2000px)"]: "calc((1080 / 24) * 1rem)"
          },
          paddingVertical: 0,
        },
      });
```
